bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/types/ArithmeticGroupFunctionTest.java

```diff
@@ -201,6 +203,44 @@ public void testSumFunction() {
         assertEquals(new DecimalType("234.95"), state);
     }
 
+    @Test
+    public void testAvgFunction_dimmers() {
+        DimmerItem dimmer1 = new DimmerItem("TestDimmer1");
+        dimmer1.setState(new PercentType(100));
+        DimmerItem dimmer2 = new DimmerItem("TestDimmer2");
+        dimmer2.setState(new PercentType(0));
+
+        items.add(dimmer1);
+        items.add(dimmer2);
+
+        function = new ArithmeticGroupFunction.Avg();
+        State decimalState = function.getStateAs(items, DecimalType.class);
+
+        assertEquals(new DecimalType("50"), decimalState);
```

Comment:
Hm, no, I do not agree.
As DecimalType, it should be a value between 0 and 1 (here 0.5), but if you get the state as PercentType, it should be 50, but this is not what your test is doing.